src/team3player/RobotPlayer.java

```diff
@@ -227,6 +239,10 @@ static boolean tryBuild(RobotType type, Direction dir) throws GameActionExceptio
      * @throws GameActionException
      */
     static boolean tryMine(Direction dir) throws GameActionException {
+        System.out.println("Miner ready? " + rc.isReady());
+        System.out.println("Miner can mine soup? " + rc.canMineSoup(dir));
+        System.out.println("Reached soup limit? " + rc.getSoupCarrying() + ":" + rc.getType().soupLimit);
+        System.out.println("Remain cooldown turns? " + rc.getCooldownTurns());
         if (rc.isReady() && rc.canMineSoup(dir)) {
```

Comment:
the isReady() is also inside of canMineSoup. All of the can<dosomething>() functions report false if the cooldown is above 1.0